src/opening-period/page/EditOpeningPeriodPage.test.tsx

```diff
@@ -0,0 +1,332 @@
+      ).toBeChecked();
+
+      expect(
+        container.querySelector('[data-test="weekdays-saturday-0-checkbox"]')
+      ).not.toBeChecked();
+
+      expect(
+        container.querySelector('[data-test="weekdays-sunday-0-checkbox"]')
+      ).not.toBeChecked();
+
+      expect(
+        container.querySelector('#time-span-state-id-0-toggle-button')
+      ).toHaveTextContent('Open');
+
+      expect(container.querySelector('#time-span-0-description')).toHaveValue(
+        'Arkena ovet menevät kiinni puolta tuntia aiemmin'
+      );
+    });
+  });
+
+  it('should render last the time-span when it has the weekdays in the end of the week', async () => {
+    let container: HTMLElement;
+
+    await act(async () => {
+      container = render(
+        <EditOpeningPeriodPage resourceId="tprek:8100" datePeriodId="1" />
+      ).container;
+
+      if (!container) {
+        throw new Error(
+          'Something went wrong in rendering of CreateNewOpeningPeriodPage'
+        );
+      }
+    });
+
+    await act(async () => {
+      expect(
+        container.querySelector('[data-test="weekdays-monday-1-checkbox"]')
+      ).not.toBeChecked();
+
+      expect(
+        container.querySelector('[data-test="weekdays-tuesday-1-checkbox"]')
+      ).not.toBeChecked();
+
+      expect(
+        container.querySelector('[data-test="weekdays-wednesday-1-checkbox"]')
+      ).not.toBeChecked();
+
+      expect(
+        container.querySelector('[data-test="weekdays-thursday-1-checkbox"]')
+      ).not.toBeChecked();
+
+      expect(
+        container.querySelector('[data-test="weekdays-friday-1-checkbox"]')
+      ).not.toBeChecked();
+
+      expect(
+        container.querySelector('[data-test="weekdays-saturday-1-checkbox"]')
+      ).toBeChecked();
+
+      expect(
+        container.querySelector('[data-test="weekdays-sunday-1-checkbox"]')
+      ).toBeChecked();
+
+      expect(
+        container.querySelector('#time-span-state-id-1-toggle-button')
+      ).toHaveTextContent('Self service');
+
+      expect(container.querySelector('#time-span-1-description')).toHaveValue(
+        'Viikonloppuna ovet menevät kiinni tuntia aiemmin'
+      );
+    });
+  });
+
+  it('should show loading indicator while loading date period', async () => {
+    jest.spyOn(api, 'getDatePeriod').mockImplementation(() => {
+      // eslint-disable-next-line @typescript-eslint/no-empty-function
+      return new Promise(() => {});
+    });
+
+    render(<EditOpeningPeriodPage resourceId="tprek:8100" datePeriodId="1" />);
+
+    await act(async () => {
+      expect(await screen.getByRole('heading', { level: 1 })).toHaveTextContent(
+        'Aukiolojakson muokkaus'
+      );
+      expect(await screen.getByText('Sivua ladataan...')).toBeInTheDocument();
+    });
+  });
+
+  it('should show error notification when loading date period fails', async () => {
+    const error: Error = new Error('Failed to load date period form options');
```

Comment:
Small improvement suggestion: I believe the error message here is little misleading, probably just due to forgotten copy-paste?

Reply:
Fixed